### ISSUE TITLE
Enable/disable individual autocorrections/symbols or font styles

### ISSUE THREAD

**Issue by rugk** [rugk/unicodify#30]
## Background

I guess I already mentioned it somewhere, where we’ve introduced groups, that a “left/right selection” or so woudl be useful, so let’s track that here.
Possibly it could also be used for re-ordering in the future…

The problems could be:

For symbols:
* I suffer from issues like https://github.com/rugk/unicodify/issues/14, but don't want to disable the whole category, but only remove one symbol.

For 
* I want to tweak my context menu and make it smaller and would like to minimize the height of my context menu.
* I want to most used things at the top

## Proposed solution

Two lists side-by-side, where you’d select one element and change the state from disabled to enabled by switching it to the other list. (I have no idea how that is called…)

## Alternatives

* I would not consider this too soon, also because it is quite hard to implement.
* Maybe a better settings UI could also be an alternative… like a huge list of checkboxes, but that is ugly…
* More fine-grained groups? I doubt that's useful…

## Additional context

Maybe one could also add your own symbols at least in the future, similarly to how word processing editors do it (Word or LibreOffice), they have such an autocorrection dialog.

But this are just ideas, nothing I’d immediately do. Just collecting some ideas and upvotes here.

**Comment by tdulcet** [comment]
> Two lists side-by-side, where you’d select one element and change the state from disabled to enabled by switching it to the other list.

I do not think this would be that difficult to implement. [Here](https://codepen.io/esausilva/pen/NYObRr) and [here](https://codepen.io/afrost/pen/EyQOxx) are a couple of examples of what I believe you are describing (they both are using JQuery, but that should be [easy to remove](https://gist.github.com/chibaye/9809e824bebbccb5b8e0764c8ee3a907#file-index-html-L35)). Your AutomaticSettings library would then only need to store the disabled list, which would require similar changes to those needed for https://github.com/TinyWebEx/AutomaticSettings/issues/15.

> Maybe one could also add your own symbols at least in the future, similarly to how word processing editors do it (Word or LibreOffice), they have such an autocorrection dialog.

For reference, here are those dialogs. LibraOffice Writer:
![image](https://user-images.githubusercontent.com/15305150/115379654-960c2500-a186-11eb-855b-828e4dedf153.png)

Microsoft Word:
![image](https://user-images.githubusercontent.com/15305150/115379763-b0460300-a186-11eb-8e80-30e56d7e6792.png)

**Comment by amourchaleur** [comment]
Would it be possible for auto superscript or subscript for text such as 10th or 1st (in thunderbird)?

**Comment by tdulcet** [comment]
> Would it be possible for auto superscript or subscript for text such as 10th or 1st (in thunderbird)?

Yes, that would certainly be possible, at least for superscript. Please feel free to create a new issue for the feature request.

**Comment by amourchaleur** [comment]
Just created the new feature request. Thanks for considering!

**Comment by Thomas-D-C** [comment]
> * it is quite hard to implement.

So, I think a list of checkboxes might be a good alternative, while waiting for something more elaborate.


> * a huge list of checkboxes, but that is ugly…

Why exactly do you find it ugly?


1. If it's because it takes up too much space compared to the usefulness:
      Have you considered "collapsible containers" (I can't remember what it's called in GTK)?
      Maybe one per group? Or one per sub-menu?

      For an unchecked group, the corresponding "collapsible containers" might be locked to "collapse". (I'm not sure about my English.)


2. If it's because it would make a code far too much repetitive:
      Have you considered code generation? :-)

      I don't know JavaScript, but I like very much code generation. :-) :-)
      Maybe you could learn me a little bit of JavaScript, and I could help you with code generation?
